pkgs/development/tools/build-managers/gradle/gen.nix

```diff
@@ -0,0 +1,66 @@
+{ version, nativeVersion, sha256, defaultJava }:
+
+{ lib, stdenv, fetchurl, unzip, makeWrapper, java ? defaultJava }:
+
+stdenv.mkDerivation rec {
+  pname = "gradle";
+  inherit version;
+
+  src = fetchurl {
+    inherit sha256;
+    url = "https://services.gradle.org/distributions/gradle-${version}-bin.zip";
+  };
+
+  dontBuild = true;
+
+  nativeBuildInputs = [ makeWrapper unzip ];
+  buildInputs = [ java ];
+
+  installPhase = ''
+    mkdir -pv $out/lib/gradle/
+    cp -rv lib/ $out/lib/gradle/
+
+    gradle_launcher_jar=$(echo $out/lib/gradle/lib/gradle-launcher-*.jar)
+    test -f $gradle_launcher_jar
+    makeWrapper ${java}/bin/java $out/bin/gradle \
+      --set JAVA_HOME ${java} \
+      --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain"
+  '';
+
+  fixupPhase = if (!stdenv.isLinux) then
+    ":"
+  else
+    let arch = if stdenv.is64bit then "amd64" else "i386";
```

Comment:
Please use dontFixup if we wich to not run anything in fixupPhase.

Reply:
Do you mean something like the following?

```diff
diff --git a/pkgs/development/tools/build-managers/gradle/gen.nix b/pkgs/development/tools/build-managers/gradle/gen.nix
index cdea1ba56cb..a10c3ab405a 100644
--- a/pkgs/development/tools/build-managers/gradle/gen.nix
+++ b/pkgs/development/tools/build-managers/gradle/gen.nix
@@ -27,24 +27,22 @@ stdenv.mkDerivation rec {
       --add-flags "-classpath $gradle_launcher_jar org.gradle.launcher.GradleMain"
   '';

-  fixupPhase = if (!stdenv.isLinux) then
-    ":"
-  else
-    let arch = if stdenv.is64bit then "amd64" else "i386";
-    in ''
-      ...
+  dontFixup = (!stdenv.isLinux);
+  fixupPhase = let arch = if stdenv.is64bit then "amd64" else "i386";
+  in ''
+    ...

```

Reply:
I think that should be resolved now.

Reply:
Fixed in a77d0d0